### PR TITLE
Added coding styles for modules and generics.

### PR DESCRIPTION
**Pull request by jacobbridges**
Most of the code style I borrowed from the TypeScript Handbook examples.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2150,7 +2188,45 @@
...
+      }
+
+    }
+
+  }
+
+  export default Automobile;
+  ```
+
+- [28.3](#28.3) <a name='28.3'></a> Modules should be Pascal case.
```
Comment (achannarasappa) [inline README.md]: This is covered in the naming conventions section: https://github.com/jacobbridges/typescript/blob/master/README.md#22.2
You could add an example for modules in this section.

**Comment by achannarasappa** [comment]
Can you also add a section about ordering inside a module? I propose we use the following order by keyword:
- var
- export var
- let
- export let
- const
- export const
- interface
- export interface
- function
- export function
- class
- export class
- module
- export module

And within each group, order alphabetically

File: README.md [hunk 2]
```diff
@@ -2134,7 +2134,45 @@
   - [26.1](#26.1) <a name='26.1'></a> Type annotations placeholder.
 
 <a name="ts-generics"></a>
-  - [26.x](#26.x) <a name='26.x'></a> Generics placeholder.
+  - [26.x](#26.x) <a name='26.x'></a> Use "T" for the type variable if only one is needed.
```
Comment (achannarasappa) [inline README.md]: Could you update the section numbers to 26.2, 26.3, etc?

**Comment by jacobbridges** [comment]
PR has been updated with your suggested edits.

**Pull request merged**